tests/unit/test_config.py

```diff
@@ -0,0 +1,170 @@
+"""Unit tests for the configuration model."""
+
+import io
+import traceback
+from typing import Type, TypeVar
+
+from yaml.parser import ParserError
+
+from ols.app.models.config import InvalidConfigurationError
+from ols.utils import config
+
+E = TypeVar("E", bound=Exception)
+
+
+def check_expected_exception(
+    yaml_stream: str, expected_exception_type: Type[E], expected_error_msg: str
+) -> None:
+    """Check that an expected exception is raised."""
+    caught_expected_exception = False
+    try:
+        config.load_config_from_stream(io.StringIO(yaml_stream))
+    except expected_exception_type as e:
+        assert str(e).startswith(expected_error_msg)
+        caught_expected_exception = True
+    except Exception:
+        print(traceback.format_exc())
+        assert False
+    assert caught_expected_exception
+
+
+def test_malformed_yaml() -> None:
+    """Check that malformed YAML is handled gracefully."""
+    check_expected_exception(
+        """[foo=123}""", ParserError, "while parsing a flow sequence"
+    )
+    check_expected_exception(
+        """foobar""", AttributeError, "'str' object has no attribute 'get'"
+    )
+    check_expected_exception(
+        """123""", AttributeError, "'int' object has no attribute 'get'"
+    )
+    check_expected_exception(
+        """12.3""", AttributeError, "'float' object has no attribute 'get'"
+    )
+    check_expected_exception(
+        """[1,2,3]""", AttributeError, "'list' object has no attribute 'get'"
```

Comment:
this seems like it's just testing the yaml parser and confirming that the config loader doesn't mask/wrap any parsing exceptions so i question the value of having so many of these tests.  To be honest we probably *should* catch+wrap parsing exceptions....we still want to show the user the parse exception, but having the config loader catch+wrap it so we can give them a sane error like `failed to load config due to yaml parse error: $original_parse_error`

Reply:
@bparees When loading configuration from a file with ols.utils.config.init_config(), the user will see, just as you say, "Failed to load config file {config_file}: {str(e)}". These tests call load_config_from_stream() directly, it doesn't handle exceptions.
I added these ^^^ tests to see how yaml.safe_load() handles 'bad' YAML. Of the four tests, only the first one is actually malformed, the other three are valid YAML but useless as OLS config and the AttributeErrors are raised by the config model code like
```
       if data is None:
            return
        v = data.get("LLMProviders") // AttributeError gets raised here
```
The AttributeError tests cover all the possible Python types we can encounter in this situation.
What can be a better option is 
```
        if not isinstance(data, dict):
            return
        v = data.get("LLMProviders")
```
which for the AttributeError tests would bail immediately and later fail validation with `InvalidConfigurationError: no LLMProviders found`

Gotta love dynamic typing. You declare `data: Optional[dict]`, check that it's not None, but it still be not a dict.